config.js

```diff
@@ -261,10 +262,30 @@ exports.updateFromArgv = function() {
         exports.fakerank = parseRank(argv.fakerank);
     }
     exports.bot_command = argv._;
+
+    exports.start_date  = new Date();
+    if (argv.rejectnewtime) {
+        const reject_date = exports.start_date;
```

Comment:
is it a const if you change the value?

Reply:
great question!!
i was waiting for someone to ask me this question because i love to explain how js works!!

in javascript objects are declared by reference.
`exports.start_date` is an instance of Date, so it has a reference like all objects

then reject_date is defined with the same reference, so it means if we modify reject_date it will also modify exports.start_date, which is a mistake that can be fixed with the spread operator:

```Js
    const reject_date = { ...exports.start_date };
```

but an object being declared as a const only means we can't change the reference of the object (memory adress), but the content can be modified as much as you want,
i made a js chapter in the js lesson here: 

@sanderland https://javascript.info/rest-parameters-spread#get-a-new-copy-of-an-array-object

Reply:
makes sense :)

Reply:
cool :)
i recommend the read on spread and rest operators i linked (the full article)

more generally, all the new ES6 features, which @Dorus is very knowledgeable about (a lot more than me at least :) )

Reply:
small appendix to this explanation:
this explanation works for typical objects, but instances of Date cannot be copied in the same way using `{ ...date }` :100: